src/crypto/common.h

```diff
@@ -5,62 +5,98 @@
 #ifndef BITCOIN_CRYPTO_COMMON_H
 #define BITCOIN_CRYPTO_COMMON_H
 
-#if defined(HAVE_CONFIG_H)
-#include "bitcoin-config.h"
-#endif
-
 #include <stdint.h>
 
-#include "compat/endian.h"
-
 uint16_t static inline ReadLE16(const unsigned char* ptr)
 {
-    return le16toh(*((uint16_t*)ptr));
+    return ((uint16_t)(ptr[0])) |
+           (((uint16_t)(ptr[1])) << 8);
 }
 
 uint32_t static inline ReadLE32(const unsigned char* ptr)
 {
-    return le32toh(*((uint32_t*)ptr));
+    return ((uint32_t)(ptr[0])) |
+           (((uint32_t)(ptr[1])) << 8) |
+           (((uint32_t)(ptr[2])) << 16) |
+           (((uint32_t)(ptr[3])) << 24);
 }
 
 uint64_t static inline ReadLE64(const unsigned char* ptr)
 {
-    return le64toh(*((uint64_t*)ptr));
+    return ((uint64_t)(ptr[0])) |
+           (((uint64_t)(ptr[1])) << 8) |
+           (((uint64_t)(ptr[2])) << 16) |
+           (((uint64_t)(ptr[3])) << 24) |
+           (((uint64_t)(ptr[4])) << 32) |
+           (((uint64_t)(ptr[5])) << 40) |
+           (((uint64_t)(ptr[6])) << 48) |
+           (((uint64_t)(ptr[7])) << 56);
 }
 
 void static inline WriteLE16(unsigned char* ptr, uint16_t x)
 {
-    *((uint16_t*)ptr) = htole16(x);
+    ptr[0] = x & 0xFF;
+    ptr[1] = (x >> 8) & 0xFF;
 }
 
 void static inline WriteLE32(unsigned char* ptr, uint32_t x)
 {
-    *((uint32_t*)ptr) = htole32(x);
+    ptr[0] = x & 0xFF;
+    ptr[1] = (x >> 8) & 0xFF;
+    ptr[2] = (x >> 16) & 0xFF;
+    ptr[3] = (x >> 24) & 0xFF;
 }
 
 void static inline WriteLE64(unsigned char* ptr, uint64_t x)
 {
-    *((uint64_t*)ptr) = htole64(x);
+    ptr[0] = x & 0xFF;
+    ptr[1] = (x >> 8) & 0xFF;
+    ptr[2] = (x >> 16) & 0xFF;
+    ptr[3] = (x >> 24) & 0xFF;
+    ptr[4] = (x >> 32) & 0xFF;
+    ptr[5] = (x >> 40) & 0xFF;
+    ptr[6] = (x >> 48) & 0xFF;
+    ptr[7] = (x >> 56) & 0xFF;
 }
 
 uint32_t static inline ReadBE32(const unsigned char* ptr)
 {
-    return be32toh(*((uint32_t*)ptr));
+    return (((uint32_t)(ptr[0])) << 24) |
+           (((uint32_t)(ptr[1])) << 16) |
+           (((uint32_t)(ptr[2])) << 8) |
+           ((uint32_t)(ptr[3]));
 }
 
 uint64_t static inline ReadBE64(const unsigned char* ptr)
 {
-    return be64toh(*((uint64_t*)ptr));
+    return (((uint64_t)(ptr[0])) << 56) |
+           (((uint64_t)(ptr[1])) << 48) |
+           (((uint64_t)(ptr[2])) << 40) |
+           (((uint64_t)(ptr[3])) << 32) |
+           (((uint64_t)(ptr[4])) << 24) |
+           (((uint64_t)(ptr[5])) << 16) |
+           (((uint64_t)(ptr[6])) << 8) |
+           ((uint64_t)(ptr[7]));
 }
 
 void static inline WriteBE32(unsigned char* ptr, uint32_t x)
 {
-    *((uint32_t*)ptr) = htobe32(x);
+    ptr[0] = (x >> 24) & 0xFF;
```

Comment:
I think the & 0xFF is redundant here, no? Unless you think there is some pipelining advantage...